File: solutions/python/triangle/1/triangle.py
def is_triangle(sides):
    return (all(side > 0 for side in sides) and
            2*max(sides) < sum(sides))

def equilateral(sides):
    return is_triangle(sides) and len(set(sides)) == 1 


def isosceles(sides):
    return is_triangle(sides) and len(set(sides)) in [1, 2]


def scalene(sides):
    return is_triangle(sides) and len(set(sides)) == 3

File: solutions/lua/triangle/1/triangle.lua
local triangle = {}

function triangle.kind(a, b, c)

    if (a <= 0 or b <= 0 or c <= 0) or
        (a + b < c or b + c < a or a + c < b) then

        error("Input Error")
    end

    if (a == b and b == c) then
        return "equilateral"
    elseif (a == b or b == c or a == c) then
        return "isosceles"
    else
        return "scalene"
    end
end

return triangle
